### ISSUE TITLE
Should be raised when iterating inside a 'ForEach' 'Action' ('foreach') style

### ISSUE THREAD

**Issue by okipol88** [DotNetAnalyzers/IDisposableAnalyzers#426]
The `IterationWarningDoesNotWork` should raise a warning:

``` cs
class IteratingUsingActionExample
{
  public void SingleItemClassicStyle()
  {
    var stream = new MemoryStream();
  }

  public void IterationWarningDoesNotWork()
  {
    Enumerable.Range(0, 2)
      .ToList()
      .ForEach(_ => new MemoryStream());
  }
}
```
